x/evm/keeper/grpc_query.go

```diff
@@ -420,7 +439,8 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 		}
 		txConfig.TxHash = ethTx.Hash()
 		txConfig.TxIndex = uint(i)
-		rsp, err := k.ApplyMessageWithConfig(ctx, msg, types.NewNoOpTracer(), true, cfg, txConfig)
+		// reset gas meter for each transaction
+		rsp, err := k.ApplyMessageWithConfig(ctx.WithGasMeter(evmostypes.NewInfiniteGasMeterWithLimit(msg.Gas())), msg, types.NewNoOpTracer(), true, cfg, txConfig)
```

Comment:
```suggestion
		ctx = ctx.WithGasMeter(evmostypes.NewInfiniteGasMeterWithLimit(msg.Gas()))
		rsp, err := k.ApplyMessageWithConfig(ctx, msg, types.NewNoOpTracer(), true, cfg, txConfig)
```
Please shorten the line to increase readability